src/EFCore.PG/Query/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -155,7 +180,100 @@ protected override Expression VisitSqlBinary(SqlBinaryExpression binary)
             }
         }
 
-        [NotNull]
+        protected virtual Expression VisitPostgresNewArray([NotNull] PostgresNewArrayExpression postgresNewArrayExpression)
+        {
+            Sql.Append("ARRAY[");
+            var first = true;
+            foreach (var initializer in postgresNewArrayExpression.Initializers)
+            {
+                if (!first)
+                {
+                    Sql.Append(",");
+                }
+
+                first = false;
+                Visit(initializer);
+            }
+
+            // Not sure if the explicit store type is necessary, but just to be sure.
+            Sql
+                .Append("]::")
+                .Append(postgresNewArrayExpression.TypeMapping.StoreType);
+
+            return postgresNewArrayExpression;
+        }
+
+        protected virtual Expression VisitPostgresBinary([NotNull] PostgresBinaryExpression binaryExpression)
+        {
+            Check.NotNull(binaryExpression, nameof(binaryExpression));
+
+            var requiresBrackets = RequiresBrackets(binaryExpression.Left);
+
+            if (requiresBrackets)
+                Sql.Append("(");
+
+            Visit(binaryExpression.Left);
+
+            if (requiresBrackets)
+                Sql.Append(")");
+
+            Sql
+                .Append(" ")
+                .Append(binaryExpression.OperatorType switch
+                {
+                    PostgresExpressionType.Contains when
+                    binaryExpression.Left.TypeMapping is NpgsqlInetTypeMapping ||
+                    binaryExpression.Left.TypeMapping is NpgsqlCidrTypeMapping => ">>",
```

Comment:
Having an indentation will improve readability in case of conditions.
```suggestion
                        binaryExpression.Left.TypeMapping is NpgsqlInetTypeMapping ||
                        binaryExpression.Left.TypeMapping is NpgsqlCidrTypeMapping => ">>",
```

Reply:
Rider is a bit weird with the indentation, but I agree with you - improved this.